pkg/app/piped/cloudprovider/ecs/client.go

```diff
@@ -0,0 +1,182 @@
+	region string
+	client *ecs.Client
+	logger *zap.Logger
+}
+
+func newClient(region, profile, credentialsFile, roleARN, tokenPath string, logger *zap.Logger) (*client, error) {
+	if region == "" {
+		return nil, fmt.Errorf("region is required field")
+	}
+
+	c := &client{
+		region: region,
+		logger: logger.Named("ecs"),
+	}
+
+	optFns := []func(*config.LoadOptions) error{config.WithRegion(region)}
+	if credentialsFile != "" {
+		optFns = append(optFns, config.WithSharedCredentialsFiles([]string{credentialsFile}))
+	}
+	if profile != "" {
+		optFns = append(optFns, config.WithSharedConfigProfile(profile))
+	}
+	if tokenPath != "" && roleARN != "" {
+		optFns = append(optFns, config.WithWebIdentityRoleCredentialOptions(func(v *stscreds.WebIdentityRoleOptions) {
+			v.RoleARN = roleARN
+			v.TokenRetriever = stscreds.IdentityTokenFile(tokenPath)
+		}))
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.Background(), optFns...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load config to create ecs client: %w", err)
+	}
+	c.client = ecs.NewFromConfig(cfg)
+
+	return c, nil
+}
+
+func (c *client) CreateService(ctx context.Context, service types.Service) (*ecs.CreateServiceOutput, error) {
+	input := &ecs.CreateServiceInput{
+		ServiceName:                   service.ServiceName,
+		Cluster:                       service.ClusterArn,
+		DeploymentConfiguration:       service.DeploymentConfiguration,
+		DeploymentController:          service.DeploymentController,
+		DesiredCount:                  &service.DesiredCount,
+		EnableECSManagedTags:          service.EnableECSManagedTags,
+		HealthCheckGracePeriodSeconds: service.HealthCheckGracePeriodSeconds,
+		LaunchType:                    service.LaunchType,
+		LoadBalancers:                 service.LoadBalancers,
+		NetworkConfiguration:          service.NetworkConfiguration,
+		PlacementConstraints:          service.PlacementConstraints,
+		PlacementStrategy:             service.PlacementStrategy,
+		PlatformVersion:               service.PlatformVersion,
+		PropagateTags:                 service.PropagateTags,
+		Role:                          service.RoleArn,
+		SchedulingStrategy:            service.SchedulingStrategy,
+		ServiceRegistries:             service.ServiceRegistries,
+		Tags:                          service.Tags,
+		TaskDefinition:                service.TaskDefinition,
+	}
+	output, err := c.client.CreateService(ctx, input)
+	if err != nil {
+		return &ecs.CreateServiceOutput{}, fmt.Errorf("failed to create ECS service %s: %w", *service.ServiceName, err)
+	}
+	return output, nil
+}
+
+func (c *client) UpdateService(ctx context.Context, service types.Service) (*ecs.UpdateServiceOutput, error) {
+	input := &ecs.UpdateServiceInput{
+		Service:                       service.ServiceName,
+		Cluster:                       service.ClusterArn,
+		DeploymentConfiguration:       service.DeploymentConfiguration,
+		DesiredCount:                  &service.DesiredCount,
```

Comment:
same as https://github.com/pipe-cd/pipe/pull/1647/files#r592092785

Reply:
@TaKO8Ki this one is missed 🙏

Reply:
Thank you! I fixed it.